src/confluent_kafka/deserializing_consumer.py

```diff
@@ -130,29 +130,83 @@ def poll(self, timeout=-1):
         if msg.error() is not None:
             raise ConsumeError(msg.error(), kafka_message=msg)
 
-        ctx = SerializationContext(msg.topic(), MessageField.VALUE)
-        value = msg.value()
+        return self._parse_deserialize_message(msg)
+
+    def consume(self, num_messages=1, timeout=-1):
+        """
+        Consume up to the number of messages specified with a timeout for each request
+
+        Args:
+            num_messages (int): The maximum number of messages to wait for.
+            timeout (float): Maximum time to block waiting for message(Seconds).
+
+        Returns:
+            :py:class:`Message` or None on timeout
+
+        Raises:
+            KeyDeserializationError: If an error occurs during key
+            deserialization.
+
+            ValueDeserializationError: If an error occurs during value
+            deserialization.
+
+            ConsumeError if an error was encountered while polling.
+
+            RuntimeError if the number of messages is less than 1
+        """
+        if num_messages < 1:
+            raise RuntimeError("The maximum number of messages must be greater than or equal to 1.")
+
+        messages = super(DeserializingConsumer, self).consume(num_messages, timeout)
+
+        if messages is None:
+            return []
+
+        deserialized_messages = []
+
+        for message in messages:
+            if message.error() is not None:
+                raise ConsumeError(message.error(), kafka_message=message)
```

Comment:
this implementation effectively causes message without error to be thrown away, which is probably not ideal.

thanks for the PR, currently re-considering the need for batch consume.

Reply:
Good point, only did this to ensure that we kept parity with the singular API, but returning the messages with the error code is likely the better idea. I removed the exception throw.